### PR TITLE
Delay RNN graph break until module call

### LINKED ISSUE #141184: Dynamo graph break on RNN/LSTM doesn't resume the function correctly.
### 🐛 Describe the bug

Dynamo purposely graph breaks on RNN, GRU, LSTMs. We expect the sub modules before and after the RNN/GRU/LSTM calls should be compiled as partial graphs, however, we found only the sub modules before the graph break got compiled, the one after doesn't.
Repro:
```
import torch
import torch.nn as nn
import torch.nn.functional as F

class MultiLayerModel(nn.Module):
    def __init__(self, input_dim, lstm_hidden_dim, conv_channels, linear_out_dim, num_classes):
        super(MultiLayerModel, self).__init__()
        
        # Convolutional layer
        self.conv = nn.Conv2d(in_channels=1, out_channels=conv_channels, kernel_size=3, stride=1, padding=1)
        
        # Layer normalization for convolution output
        self.conv_layer_norm = nn.LayerNorm([conv_channels, input_dim, input_dim])
        
        # LSTM layer
        self.lstm = nn.LSTM(input_dim, lstm_hidden_dim, batch_first=True, bidirectional=True)
        
        # Fully connected linear layers
        self.fc1 = nn.Linear(lstm_hidden_dim * 2, linear_out_dim)  # LSTM is bidirectional, so hidden_dim * 2
        self.fc2 = nn.Linear(linear_out_dim, num_classes)
        
        # Layer normalization for linear layers
        self.linear_layer_norm = nn.LayerNorm(linear_out_dim)
        
    def forward(self, x):
        # Convolutional layer expects 4D input: (batch_size, channels, height, width)
        x = self.conv(x)
        x = F.relu(x)
        x = self.conv_layer_norm(x)
        
        # Flatten spatial dimensions for the LSTM (assuming square input images)
        batch_size, channels, height, width = x.size()
        x = x.view(batch_size, channels * height, width)  # Prepare for LSTM: (batch, seq_len, features)
        
        # LSTM expects input: (batch_size, seq_len, input_dim)
        x, _ = self.lstm(x)
        
        # Fully connected layers
        x = F.relu(self.fc1(x[:, -1, :]))  # Use the output from the last time step
        x = self.linear_layer_norm(x)
        x = self.fc2(x)
        
        return x
    
input_dim = 28
lstm_hidden_dim = 128
conv_channels = 16
linear_out_dim = 64
num_classes = 10

model = MultiLayerModel(input_dim, lstm_hidden_dim, conv_channels, linear_out_dim, num_classes)
print(model)

model = torch.compile(model)
# Dummy input: Batch of grayscale images (batch_size, channels, height, width)
dummy_input = torch.randn(8, 1, 28, 28)
output = model(dummy_input)
print("Output shape:", output.shape)
```
Run this script with ```TORCH_LOGS=+dynamo```, you will find there is only one compiled graph which includes the sub module befroe line ```x = x.view(batch_size, channels * height, width)```.

### Versions

N/A

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.15.0) (oldest at bottom):
* __->__ #186614

Dynamo intentionally graph breaks on nn.RNN, nn.GRU, and nn.LSTM when
allow_rnn is disabled. The old implementation raised the unsupported error
while wrapping the recurrent module attribute. For LOAD_ATTR/LOAD_METHOD
bytecode this goes through the generic step graph-break path, which can
compile the prefix but cannot build a continuation for the call site, so the
rest of the frame runs eagerly and post-RNN code is never compiled.

Return a DelayGraphBreakVariable for unsupported recurrent modules instead.
This keeps the attribute load traceable enough to reach the CALL instruction,
where Dynamo's regular call graph-break handling can run the recurrent module
eagerly and resume tracing the continuation. This matches the existing delayed
break pattern used for disabled optimized modules and avoids changing the
experimental allow_rnn=True behavior.

Add regression coverage for RNN, GRU, and LSTM modules that verifies Dynamo
compiles both the pre-recurrent and post-recurrent regions.

Fixes #141184
Generated by my agent

Test Plan:
- python test/dynamo/test_modules.py -k test_rnn_graph_break_resumes_after_call
- python test/export/test_export.py -k test_export_dynamo_config
- python test/dynamo/test_modules.py -k unsupportedmodule
- lintrunner -a

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @chauhang @amjames @jataylo @azahed98

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says I addressed the inductor TorchBench CI failures by updating the expected graph-break baselines for demucs, doctr_reco_predictor, and tts_angular. These extra breaks are expected now that the RNN/GRU/LSTM graph break is delayed to the module call and Dynamo can resume far enough to reach additional recurrent calls; accuracy still passed. The torchtitan process-group failure looked unrelated and the new CI run is pending.

**Comment by jansel** [comment]
My agent says I fixed the remaining Dr. CI new failures by updating the inductor TorchBench training graph-break baselines for demucs and tts_angular. These are the same expected fallout from delaying the RNN/GRU/LSTM graph break to the module call: Dynamo now resumes further and reaches additional recurrent calls while accuracy still passes.

**Review by mlazos — APPROVED** [review]
